### ISSUE TITLE
undeploy failed: Unable connect to kubernetes

### ISSUE THREAD

**Issue by parayacr** [dotnet/tye#1303]
### Describe the bug
I was following the tutorial (https://github.com/dotnet/tye/blob/main/docs/tutorials/hello-tye/01_deploy.md) and everything is working fine except when trying to run `tye undeploy`. For more context, I used the dockerhub registry and minikube (minikube version: v1.25.1).

### To Reproduce
Following the tutorial all the way to the deploy part where it explains how to undeploy the application.

```
Loading Application Details...
Drats! 'undeploy' failed:
        Unable connect to kubernetes.

Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'NotFound'
   at k8s.Kubernetes.ListNamespacedIngressWithHttpMessagesAsync(String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, String pretty, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Tye.UndeployHost.ExecuteUndeployAsync(OutputContext output, ApplicationBuilder application, String namespace, Boolean interactive, Boolean whatIf) in /_/src/tye/UndeployHost.cs:line 143
```

### Further technical details

- `tye version`
0.10.0-alpha.21420.1+7cec0655cc474aee8d3c3a3d180678985209d439
- `tye undeploy -v debug`
```
Loading Application Details...
Restoring and evaluating projects
Resolved metadata for service frontend at C:\Users\araya\source\repos\distributed_apps\tye\microservice\frontend\obj\Debug\net6.0\MicrosoftTye.ProjectMetadata.txt
Resolved metadata for service backend at C:\Users\araya\source\repos\distributed_apps\tye\microservice\backend\obj\Debug\net6.0\MicrosoftTye.ProjectMetadata.txt
Restore and project evaluation took: 1229.0876ms
Found application version: 1.0.0
RunCommand=C:\Users\araya\source\repos\distributed_apps\tye\microservice\frontend\bin\Debug\net6.0\frontend.exe
RunArguments=
TargetPath=C:\Users\araya\source\repos\distributed_apps\tye\microservice\frontend\bin\Debug\net6.0\frontend.dll
PublishDir=bin\Debug\net6.0\publish\
AssemblyName=frontend
IntermediateOutputPath=obj\Debug\net6.0\
Found target framework: net6.0
Parsed target framework name: net
Parsed target framework version: 6.0
Found shared frameworks: Microsoft.NETCore.App, Microsoft.AspNetCore.App
IsAspNet=True
Evaluation Took: 3.3069ms
Found application version: 1.0.0
RunCommand=C:\Users\araya\source\repos\distributed_apps\tye\microservice\backend\bin\Debug\net6.0\backend.exe
RunArguments=
TargetPath=C:\Users\araya\source\repos\distributed_apps\tye\microservice\backend\bin\Debug\net6.0\backend.dll
PublishDir=bin\Debug\net6.0\publish\
AssemblyName=backend
IntermediateOutputPath=obj\Debug\net6.0\
Found target framework: net6.0
Parsed target framework name: net
Parsed target framework version: 6.0
Found shared frameworks: Microsoft.NETCore.App, Microsoft.AspNetCore.App
IsAspNet=True
Evaluation Took: 1.382ms
Querying services
Found 3 matching services
Querying deployments
Found 3 matching deployments
Querying secrets
Found 1 matching secrets
Querying ingresses
Failed to query ingress.
Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'NotFound'
   at k8s.Kubernetes.ListNamespacedIngressWithHttpMessagesAsync(String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, String pretty, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Tye.UndeployHost.ExecuteUndeployAsync(OutputContext output, ApplicationBuilder application, String namespace, Boolean interactive, Boolean whatIf) in /_/src/tye/UndeployHost.cs:line 143
Drats! 'undeploy' failed:
        Unable connect to kubernetes.

Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'NotFound'
   at k8s.Kubernetes.ListNamespacedIngressWithHttpMessagesAsync(String namespaceParameter, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, String pretty, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Tye.UndeployHost.ExecuteUndeployAsync(OutputContext output, ApplicationBuilder application, String namespace, Boolean interactive, Boolean whatIf) in /_/src/tye/UndeployHost.cs:line 143
```
- Current tye.yml
```
name: microservice
registry: myregistry
services:
- name: frontend
  project: frontend/frontend.csproj
- name: backend
  project: backend/backend.csproj
- name: redis
  image: redis
  bindings:
  - port: 6379
    connectionString: "${host}:${port}" 
- name: redis-cli
  image: redis
  args: "redis-cli -h redis MONITOR"
```
- The platform (Linux/macOS/Windows): 
```
Edition	Windows 11 Education
Version	21H2
Installed on	‎9/‎9/‎2021
OS build	22000.466
```

**Comment by JhansiSai** [comment]
Try to execute this command in your Azure CLI
az aks get-credentials --name <cluster-name> --resource-group <resource-group-name>

**Comment by philliphoff** [comment]
@parayacr Could you try with the (just released) 0.11.0 release?  We pulled forward the version of Kubernetes SDK used by Tye, which might (hopefully) resolve issues using Tye with recent flavors of Kubernetes API.  (The previous version of SDK was very old.)

**Comment by parayacr** [comment]
Thanks for the suggestion @philliphoff, it is now working with the latest 0.11.0 release.
